broadinstitute/single_cell_portal_core
# (SCP-3602) Upload wizard react refactor

SCP-3602.  This puts the initial code for a file upload wizard using React/API instead of Rails forms.  It's behind a feature flag for ease of future release, but this merge is to an integration branch since I don't think this is ready for even staging-level testing yet.  Also, this is altering the manage_study API in a non-feature-flagged way, so we will want to test mange_study prior to releasing this, even behind a feature flag.

Currently, the cluster and image file upload forms are working, everything else is not yet implemented. See SCP-3655 for remaining work

![image](https://user-images.githubusercontent.com/2800795/133152788-8f5d4d6a-1eea-4aef-b22f-455c791bb924.png)


TO TEST:
1. Run migration script, then set `react_upload_wizard` feature flag to true in your environment/user
2. Go to the upload wizard, and select Cluster or Image file
3. Confirm you can edit, create, save, and delete files, and they get uploaded to firecloud.
